src/Microsoft.Azure.CosmosEventSourcing/Events/AtomicEvent.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.CosmosEventSourcing.Aggregates;
+using Microsoft.Azure.CosmosEventSourcing.Stores;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.CosmosEventSourcing.Events;
+
+/// <summary>
+/// An event that is used to pass between <see cref="IAggregateRoot"/>'s and the <see cref="IEventStore{TEventItem}"/>
+/// <remarks>This is used to provide optimistic concurrency control for batches of <see cref="DomainEvent"/>'s.</remarks>
+/// </summary>
+public record AtomicEvent : DomainEvent
```

Comment:
@IEvangelist any ideas on a better name for this? It's basically a marker that needs to be updated with every batch of new events. Struggle with naming lol

Reply:
I dotn't mind `AtomicEvent`, another name might be `BatchEvent`?

Reply:
If that makes sense to you happy to keep it as is 😃